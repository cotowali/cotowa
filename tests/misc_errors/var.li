var i = 0
var i = 0
var n: int = 's'
n = 's'

var (t1, t2): (int, int) = ('a', 'b')
(i, n) = ('a', 'b')

var _ = 10
echo(_)

fn f() {
  n = '10'
  (i, n) = ('a', 'b')
}

fn mismatched_variables_count() {
  var (a, b, c): (int, int)
}

fn assign_to_const() {
  const (a, b) = (0, 1)
  a = 10
  b = 10
  (a, b) = 10

  const arr = [[0, 1], [2, 3]]
  arr = 0
  arr[0] = 0
  arr[0][0] = 0
}
