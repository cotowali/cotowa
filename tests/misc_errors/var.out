error: var.li 2,5: variable `i` is already defined
    2 | var i = 0
      |     ^
error: var.li 3,5: mismatched types: `int` (expected) and `string` (found)
    3 | var n: int = 's'
      |     ^^^^^^^^^^^^
error: var.li 4,1: mismatched types: `int` (expected) and `string` (found)
    4 | n = 's'
      | ^^^^^^^
error: var.li 6,5: mismatched types: `(int, int)` (expected) and `(string, string)` (found)
    6 | var (t1, t2): (int, int) = ('a', 'b')
      |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: var.li 7,1: mismatched types: `(int, int)` (expected) and `(string, string)` (found)
    7 | (i, n) = ('a', 'b')
      | ^^^^^^^^^^^^^^^^^^^
error: var.li 10,9: variable `_` is not defined
   10 | var _ = _ + _
      |         ^
error: var.li 10,13: variable `_` is not defined
   10 | var _ = _ + _
      |             ^
error: var.li 13,3: cannot assign to variables outside of current function
   13 |   n = '10'
      |   ^^^^^^^^
error: var.li 14,3: cannot assign to variables outside of current function
   14 |   (i, n) = ('a', 'b')
      |   ^^^^^^^^^^^^^^^^^^^
error: var.li 18,7: expected 2 variables, but found 3 variables
   18 |   var (a, b, c): (int, int)
      |       ^^^^^^^^^
error: var.li 25,3: mismatched types: `(int, int)` (expected) and `int` (found)
   25 |   (a, b) = 10
      |   ^^^^^^^^^^^
error: var.li 28,3: mismatched types: `[][]int` (expected) and `int` (found)
   28 |   arr = 0
      |   ^^^^^^^
error: var.li 28,3: cannot assign to constant variable
   28 |   arr = 0
      |   ^^^^^^^
error: var.li 29,3: cannot assign to constant variable
   29 |   arr[0] = 0
      |   ^^^^^^^^^^
error: var.li 30,3: cannot assign to constant variable
   30 |   arr[0][0] = 0
      |   ^^^^^^^^^^^^^
